internal/impl/ockam/input_ockam_kafka_franz.go

```diff
@@ -0,0 +1,193 @@
+		Field(kafka.SaslField()).
+		Field(service.NewBoolField("multi_header").Description("Decode headers into lists to allow handling of multiple values with the same key").Default(false).Advanced()).
+		Field(service.NewBatchPolicyField("batching").
+			Description("Allows you to configure a [batching policy](/docs/configuration/batching) that applies to individual topic partitions in order to batch messages together before flushing them for processing. Batching can be beneficial for performance as well as useful for windowed processing, and doing so this way preserves the ordering of topic partitions.").
+			Advanced()).
+		LintRule(`
+let has_topic_partitions = this.topics.any(t -> t.contains(":"))
+root = if $has_topic_partitions {
+  if this.consumer_group.or("") != "" {
+    "this input does not support both a consumer group and explicit topic partitions"
+  } else if this.regexp_topics {
+    "this input does not support both regular expression topics and explicit topic partitions"
+  }
+}
+`)
+}
+
+func init() {
+	err := service.RegisterBatchInput(
+		"ockam_kafka_franz", ockamFranzKafkaInputConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchInput, error) {
+			i, err := newOckamKafkaFranzInput(conf, mgr)
+			if err != nil {
+				return nil, err
+			}
+
+			r, err := service.AutoRetryNacksBatchedToggled(conf, i)
+			if err != nil {
+				return nil, err
+			}
+
+			return conf.WrapBatchInputExtractTracingSpanMapping("ockam_kafka_franz", r)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+//------------------------------------------------------------------------------
+
+func newOckamKafkaFranzInput(conf *service.ParsedConfig, mgr *service.Resources) (*ockamFranzKafkaInput, error) {
+	// Create a franzKafkaReader instance that the ockamFranzKafkaInput will use
+	franzKafkaReader, err := kafka.NewFranzKafkaReaderFromConfig(conf, mgr)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeName, err := conf.FieldString("node_name")
+	if err != nil {
+		nodeName = "kafka-input-" + strconv.Itoa(rand.Int())
+	}
+
+	nodePort, err := conf.FieldString("node_port")
+	if err != nil {
+		// If the node_port is not set, find a free port
+		listener, err := net.Listen("tcp", ":0")
+		if err != nil {
+			return nil, err
+		}
+		nodePort = strconv.Itoa(listener.Addr().(*net.TCPAddr).Port)
+		_ = listener.Close()
+	}
+
+	// Use the first "seed_brokers" field item as the bootstrap server
+	seedBrokers, err := conf.FieldStringList("seed_brokers")
+	if err != nil {
+		return nil, err
+	}
+	bootstrapServer := strings.Split(seedBrokers[0], ",")[0]
+
+	kafkaInletAddress, err := conf.FieldString("inlet_address")
+	if err != nil {
+		listener, err := net.Listen("tcp", ":0")
+		if err != nil {
+			return nil, err
+		}
+		port := strconv.Itoa(listener.Addr().(*net.TCPAddr).Port)
+		kafkaInletAddress = "0.0.0.0:" + port
+		_ = listener.Close()
+	}
+
+	_, tls, err := conf.FieldTLSToggled("tls")
+	if err != nil {
+		tls = false
+	}
+
```

Comment:
I guess ultimately what we should do is modify the `seed_brokers` config we pass to the underling plugin,  to point to our  `kafkaInletAddress`  instead right?. 

 And that `kafkaInletAddress` , _might_  not be on configuration at all if we figure out how to choose a random unused port we can bind to.  (although this likely can wait and be an improvement for latter)

Reply:
> I guess ultimately what we should do is modify the seed_brokers config we pass to the underling plugin, to point to our kafkaInletAddress instead right?.

Yeah, that's the last point to fix now. My first goal was to use our plugin and check that everything was working normally. I'll push new changes with this.

> And that kafkaInletAddress , might not be on configuration at all if we figure out how to choose a random unused port we can bind to. (although this likely can wait and be an improvement for latter)

I'll take a look at this as well.

Reply:
I fixed the configs issues. Regarding the `kafkaInletAddress`, I've removed it, but I also had to add a new argument `broker_address` to set the raw kafka address (defaults to `localhost:9092`).

So we now how the `broker_address` that we use on our side to set up the outlet, and the `seed_brokers` that is used by the underlying kafka plugin (which now points to our inlet address). We could improve the names of our arguments probably.